frappe/database/database.py

```diff
@@ -1276,6 +1279,49 @@ def bulk_insert(
 		while value_chunk := tuple(itertools.islice(value_iterator, chunk_size)):
 			query.insert(*value_chunk).run()
 
+	def bulk_update(
+		self,
+		dt,
+		dn,
+		field,
+		val=None,
+		modified=None,
+		modified_by=None,
+		update_modified=False,
+		debug=False,
+		chunk_size=100_000,
+	):
+		"""
+		Extention of `set_value` method to update multiple records at a time in a performant way.
+
+		:param dt: DocType name.
+		:param dn: Document name for updating single record or filters for updating many records.
+		:param field: Property / field name or dictionary of values to be updated
+		:param value: Value to be updated.
+		:param modified: Use this as the `modified` timestamp.
+		:param modified_by: Set this user as `modified_by`.
+		:param update_modified: default False. Set as true, if you want to update the timestamp.
+		:param debug: Print the query in the developer / js console.
+		:param chunk_size: Number of records to be updated in one query.
+		"""
+
+		to_update = self._get_update_dict(
+			field, val, modified=modified, modified_by=modified_by, update_modified=update_modified
+		)
+
+		while self.exists(dt, dn):
+			self.set_value(
+				dt,
+				dn,
+				to_update,
+				modified=modified,
+				modified_by=modified_by,
+				update_modified=update_modified,
+				debug=debug,
+				limit=chunk_size,
+			)
+			self.commit()
```

Comment:
This is horrible for things like `rename` partial renames have been source of many bugs in past.

Reply:
We might as well just get rid of running DB transaction and then writes will be much faster :woozy_face:

Reply:
Logically, all validations are complete before the update is started. After validations, it just updates the names, and then `after_rename` is triggered.

On the safer side, the commit can be done before `after_rename`. Other than that, everything should be in our control.

Reply:
Or, `after_rename` can be part of try-except.
However, I feel it's not a good idea to disallow updates/renames just because of the number of records it can affect. This will change from site to site.

Reply:
Things fall with all sorts of failure modes, getting rid of atomicity is a big nono IMO. A failed company rename practically bricks the site if it can't be rolled back.